src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -144,11 +151,17 @@ def set_opts(self, curl, url, params, headers,
         curl.setopt(pycurl.CONNECTTIMEOUT, self.connecttimeout)
         curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
         curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
+
+        # also accepts encoding/compression algorithms
+        if headers and headers.get("Accept-Encoding"):
+            curl.setopt(pycurl.ENCODING, headers["Accept-Encoding"])
```

Comment:
this may fail if headers data-type will not be properly set. I suggest get extract headers, verify proper data-type and then call setopt.

Reply:
Did you mean the "Content-Type" header? There is no such "data-type".
However, honestly speaking I simply used the same header logic that we have been using for httplib requests, in the super classes Requests/Service.

Reply:
So, ask yourself a question what is a data type of `headers["Accept-Encoding"]` Is it an integer, dict, list? You're passing it to setopt and if data-type of return value will match the expected one you'll get an error here. My suggestion is to do this:
```
value = headers.get("Accept-Encoding" ,"") # here you should put proper default (data-type value), I expect it should be string
if isinstance(value, str):
   curl.setopt(pycurl.ENCODING, value)
```

Reply:
Gotcha! I thought you were speaking about the http request itself and its composition.
Yes, input data sanitization can be done. I'll update it.